frontend/src/lib/components/ui/button/button.svelte

```diff
@@ -55,7 +61,29 @@
 	}: ButtonProps = $props();
 </script>
 
-{#if href}
+{#if framed}
+	<WavyFrame class={href ? 'wavy-frame-button' : ''}>
+		{#if href}
+			<a
+				bind:this={ref}
+				class={cn(buttonVariants({ variant, size, className }))}
+				{href}
+				{...restProps}
+			>
+				{@render children?.()}
+			</a>
+		{:else}
+			<button
+				bind:this={ref}
+				class={cn(buttonVariants({ variant, size, className }))}
+				{type}
+				{...restProps}
+			>
+				{@render children?.()}
+			</button>
+		{/if}
+	</WavyFrame>
+{:else if href}
```

Comment:
It looks like the `framed` parameter just wraps the button in a WaveFrame component, wouldn't you rather just wrap it in the call sites instead of changing the button like this?